code/model/modelanimation_moveables.cpp

```diff
@@ -0,0 +1,134 @@
+		if(args.size() < 3){
+			Error(LOCATION,"Tried updating moveable rotation with too few (%d of 3) arguments!", (int) args.size());
+			return;
+		}
+
+		auto& anim = m_instances[pmi->id].animation;
+
+		try {
+			auto p = linb::any_cast<int>(args[0]), h = linb::any_cast<int>(args[1]), b = linb::any_cast<int>(args[2]);
+
+
+			auto& sequence = *std::static_pointer_cast<ModelAnimationSegmentSerial>(anim->m_animation);
+			auto& setOrientation = *std::static_pointer_cast<ModelAnimationSegmentSetOrientation>(sequence.m_segments[0]);
+			auto& rotation = *std::static_pointer_cast<ModelAnimationSegmentRotation>(sequence.m_segments[1]);
+			
+			ModelAnimationSubmodelBuffer buffer;
+			buffer[rotation.m_submodel].data.orientation = IDENTITY_MATRIX;
+			//Will now only contain the delta of the rotation
+			sequence.m_segments[1]->calculateAnimation(buffer, anim->getTime(pmi->id), pmi->id);
+			
+			anim->forceRecalculate(pmi);
+			
+			matrix startOrient = setOrientation.m_targetOrientation;
+			matrix newOrient;
+			//Apply rotation to old default state
+			vm_matrix_x_matrix(&newOrient, &startOrient, &buffer[rotation.m_submodel].data.orientation);
+			setOrientation.m_targetOrientation = newOrient;
+			
+			rotation.m_targetAngle = angles{fl_radians((float)p), fl_radians((float)b), fl_radians((float)h)};
+		}
+		catch(const linb::bad_any_cast& e){
+			Error(LOCATION, "Failed to parse update for rotation moveable: %s", e.what());
+		}
+	}
+
+	void ModelAnimationMoveableRotation::initialize(ModelAnimationSet* parentSet, polymodel_instance* pmi) {
+		auto& anim = m_instances[pmi->id].animation;
+
+		anim = std::shared_ptr<ModelAnimation>(new ModelAnimation(false, false, true, parentSet));
+
+		matrix orient;
+		vm_angles_2_matrix(&orient, &m_defaultPosOrient);
+		
+		auto submodel = parentSet->getSubmodel(m_submodel);
+		
+		auto sequence = std::shared_ptr<ModelAnimationSegmentSerial>(new ModelAnimationSegmentSerial());
+		sequence->addSegment(std::shared_ptr<ModelAnimationSegment>(new ModelAnimationSegmentSetOrientation(submodel, orient, true)));
+		sequence->addSegment(std::shared_ptr<ModelAnimationSegment>(new ModelAnimationSegmentRotation(submodel, m_defaultPosOrient, m_velocity, optional<float>(), m_acceleration, true)));
+		
+		anim->setAnimation(sequence);
+
+		anim->start(pmi,ModelAnimationDirection::FWD);
+	}
+	
+	std::shared_ptr<ModelAnimationMoveable> ModelAnimationMoveableRotation::parser() {
+		required_string("+Angle:");
+		angles angle;
+		stuff_angles_deg_phb(&angle);
+
+		required_string("+Velocity:");
+		angles velocity;
+		stuff_angles_deg_phb(&velocity);
+
+		optional<angles> acceleration;
+		if (optional_string("+Acceleration:")) {
+			angles parse;
+			stuff_angles_deg_phb(&parse);
+			acceleration = parse;
+		}
+
+		auto submodel = ModelAnimationParseHelper::parseSubmodel();
+		if(submodel == nullptr)
+			error_display(1, "Orientation Rotation has no target submodel!");
```

Comment:
Another question about error vs. warning

Reply:
That is similar to the above case. If this occurs, the entire moveable is not valid and cannot be created. Since this would incur followup warnings that might confuse some people, this should be a hard error.
Though, similar to the case above, if similar error harshness is intended, the "Moveable not found" case should be a hard error as well

Reply:
No change to the error is needed then.  If you can think of a more descriptive message, elaboration would be good; otherwise this is fine as-is.